### PR TITLE
EVG-13647 update code-server

### PR DESCRIPTION
**Pull request by ybrill**
Changes necessitated by changes to how code-server is structured as of [v3.10.2](https://github.com/cdr/code-server/releases/tag/v3.10.2).
This is the script that produced the latest (6-25) version of code-server now available in s3.

Other changes include:
- updating the favicon and icon to use the new Evergreen logo
- updating the extensions to the highest version compatible with code-server release v3.10.2
- fixing the s3 upload permissions

### CODE REVIEW

File: icons/favicon-dark-support.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        </g>
+    </g>
+</svg>
```
Comment (hhoke) [inline icons/favicon-dark-support.svg:16]: one of the online svg viewers I put this into said it wasn't valid svg, and the other failed to visualize. I don't know much about svg but might be cautious about the `<style>` block

File: icons/favicon.svg [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+        </g>
+    </g>
+</svg>
```
Comment (hhoke) [inline icons/favicon.svg:9]: Conversely, this svg works in https://www.svgviewer.dev/.

**Review by hhoke — APPROVED** [review]
Not sure where nameShort is referenced in product_json but assuming that's fine

**Comment by ybrill** [comment]
nameShort is referenced [here](https://github.com/cdr/code-server/blob/e9d4f877f9465c554cfe50692c6f5010535776da/lib/vscode/product.json#L2)

File: icons/favicon-dark-support.svg (hunk 1 above)
Comment (ybrill) [reply icons/favicon-dark-support.svg:16]: Thanks for pointing this out! I fixed the svg and now it changes (to white) when dark mode is enabled

**Pull request merged**